### DISCUSSION TITLE
Make RepublishMessageRecoverer.prefixedOriginalRoutingKey protected instead of private

### DISCUSSION THREAD

**Discussion by jakubvistula** [rabbitmq/rabbitmq-server#3624] [Ideas]
Hello,

How do you find idea of making String RepublishMessageRecoverer.prefixedOriginalRoutingKey protected instead of private?
It would make if possible to override prefixedOriginalRoutingKey to send back with different routing key depending on message.

My case is I would like to add custom headers to message that is moved to DLQ (x-dead-letter-exchange) that is associated with a queue.
RejectAndDontRequeueRecoverer would be fine, but I can't add custom headers there.
RepublishMessageRecoverer is good for headers, but it sends back to wrong exchange - I would like to make it behave the same as RejectAndDontRequeueRecoverer about used DLQ/exchange.
Now I will have to override whole recover() with copy paste and write custom private prefixedOriginalRoutingKey.
If prefixedOriginalRoutingKey were protected, it would be sufficient to override it without necessity of copy-paste recover().

Regards

**Comment by Anatoly4444** [comment]
@jakubvistula RepublishMessageRecoverer has additionalHeaders method. Perhaps it's not right place to ask this question

**Comment by lukebakken** [comment]
@jakubvistula - you're asking a question about Spring AMQP. Please ask here: https://github.com/spring-projects/spring-amqp/discussions
